src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -193,6 +213,94 @@ MultiLogicalPlanCreate(Query *originalQuery, Query *queryTree,
 
 
 /*
+ * ReplaceExternParamsOnOriginalQuery replaces the external parameters that appears
+ * on the query with the corresponding entries on the boundParams.
+ *
+ * Note that this function is inspired by eval_const_expr() on Postgres. We failed to
+ * use that function given that it requires access to PlannerInfo and does evaluations
+ * on other parts of the query.
```

Comment:
For example, if the query includes a `Const` expression such as:
```SQL
SELECT 3+5, user_id FROM ..
```

`eval_const_expr ()` replaces that with
```SQL
SELECT 8, user_id FROM ..
```

which we don't want because it is in the original query? (Similarly for immutable functions etc)